tfhe/src/safe_deserialization.rs

```diff
@@ -154,129 +154,109 @@ mod test_shortint {
 
 #[cfg(all(test, feature = "integer"))]
 mod test_integer {
-    use crate::conformance::{ListSizeConstraint, ParameterSetConformant};
+    use crate::conformance::ListSizeConstraint;
     use crate::high_level_api::{generate_keys, ConfigBuilder};
-    use crate::integer::parameters::RadixCiphertextConformanceParams;
     use crate::prelude::{FheDecrypt, FheTryEncrypt};
     use crate::safe_deserialization::{safe_deserialize_conformant, safe_serialize};
-    use crate::shortint::parameters::{
-        PARAM_MESSAGE_2_CARRY_2_KS_PBS, PARAM_MESSAGE_3_CARRY_3_KS_PBS,
+    use crate::shortint::parameters::PARAM_MESSAGE_1_CARRY_1_KS_PBS;
+    use crate::{
+        CompactFheUint8, CompactFheUint8ConformanceParams, CompactFheUint8List,
+        CompactFheUint8ListConformanceParams, CompactPublicKey,
     };
-    use crate::{CompactFheUint8, CompactFheUint8List, CompactPublicKey};
-
     #[test]
     fn safe_desererialization_ct() {
-        let config = ConfigBuilder::default().build();
+        let (client_key, server_key) = generate_keys(ConfigBuilder::default().build());
 
-        let (client_key, _server_key) = generate_keys(config);
+        let (_, incompatible_server_key) = generate_keys(
+            ConfigBuilder::with_custom_parameters(PARAM_MESSAGE_1_CARRY_1_KS_PBS, None).build(),
+        );
 
         let public_key = CompactPublicKey::new(&client_key);
 
         let msg = 27u8;
 
-        let num_blocks_per_integer = 4;
-
         let ct = CompactFheUint8::try_encrypt(msg, &public_key).unwrap();
 
-        assert!(
-            ct.is_conformant(&RadixCiphertextConformanceParams::from_pbs_parameters(
-                PARAM_MESSAGE_2_CARRY_2_KS_PBS,
-                num_blocks_per_integer
-            ))
-        );
-
         let mut buffer = vec![];
 
         safe_serialize(&ct, &mut buffer, 1 << 40).unwrap();
 
+        let params = CompactFheUint8ConformanceParams::from(&incompatible_server_key);
```

Comment:
Could be built directly from incompatible parameters

Reply:
done